File: Modules/SEDO/EMS/SIE.Web.EMS/Checks/Plans/Commands/ExeChangeSparePartCommand.cs
using SIE.Domain;
using SIE.Domain.Validation;
using SIE.EMS.Checks;
using SIE.EMS.Checks.Plans;
using SIE.Web.Command;
using System;
using System.Collections.Generic;
using System.Linq;

namespace SIE.Web.EMS.Checks.Plans.Commands
{
    /// <summary>
    /// 更换备件 命令
    /// </summary>
    [JsCommand("SIE.Web.EMS.Checks.Plans.Commands.ExeChangeSparePartCommand")]
    public class ExeChangeSparePartCommand : ViewCommand
    {
        /// <summary>
        /// 执行更换备件命令
        /// </summary>
        /// <param name="args">args</param>
        /// <param name="scope">scope</param>
        /// <returns>执行结果</returns>
        protected override object Excute(ViewArgs args, string scope)
        {
            var datas = args.Data.ToJsonObject<List<CheckPlanSparePart>>();
            if (datas.Count > 0)
                RT.Service.Resolve<CheckPlanController>().UIChangeCheckPlanSparePart(datas);
            else
                throw new ValidationException("未添加更换单明细".L10N());

            return true;
        }
    }
}

File: Modules/SEDO/EMS/SIE.Web.EMS/Checks/Plans/Commands/ExeChangeSparePartCommand.js
SIE.defineCommand('SIE.Web.EMS.Checks.Plans.Commands.ExeChangeSparePartCommand', {
    extend: 'SIE.cmd.Save',
    meta: { text: "更换", group: "edit", iconCls: "icon-Check" },
    canExecute: function (view) {
        var entity = view.getParent().getCurrent();
        return (entity.getExeState() == 0 || entity.getExeState() == 4);
    },
    execute: function (view) {
        var commandInfo = {
            command: "SIE.Web.EMS.Checks.Plans.Commands.ExeChangeSparePartCommand",
            entityType: "SIE.EMS.Checks.Plans.CheckPlanSparePart",
            parentType: "SIE.EMS.Checks.Plans.ViewModels.CheckPlanViewModel",
            moduleName: "点检计划维护",
            childModuleName: "",
            commandName: "更换",
        }
        var datas = [];
        Ext.each(view.getData().data.items, function (data) { datas.push(data.data); });
        view.execute({
            data: datas,
            logInfo: commandInfo,
            success: function (res) {
                SIE.Msg.showInstantMessage('备件更换成功'.t());
                view.loadChildData(true);
            }
        });
    }
});
